src/cascadia/TerminalSettingsModel/ActionArgsMagic.h

```diff
@@ -54,50 +54,105 @@ struct InitListPlaceholder
 // expanded. Pretty critical for tracking down extraneous commas, etc.
 
 // Property definitions, and JSON keys
-#define DECLARE_ARGS(type, name, jsonKey, required, ...)    \
-    static constexpr std::string_view name##Key{ jsonKey }; \
+#define DECLARE_ARGS(type, name, jsonKey, required, typeHint, ...) \
+    static constexpr std::string_view name##Key{ jsonKey };        \
     ACTION_ARG(type, name, ##__VA_ARGS__);
 
 // Parameters to the non-default ctor
-#define CTOR_PARAMS(type, name, jsonKey, required, ...) \
+#define CTOR_PARAMS(type, name, jsonKey, required, typeHint, ...) \
     const type &name##Param,
 
 // initializers in the ctor
-#define CTOR_INIT(type, name, jsonKey, required, ...) \
+#define CTOR_INIT(type, name, jsonKey, required, typeHint, ...) \
     _##name{ name##Param },
 
+// Expands to `+1` for every arg in the list
+#define COUNT_ONE(type, name, jsonKey, required, typeHint, ...) +1
+
+// Arg count macro
+#define ARG_COUNT(argsMacro) (0 argsMacro(COUNT_ONE))
+
+#define ARG_DESC_STRINGIFY2(x) #x
+#define ARG_DESC_STRINGIFY(x) ARG_DESC_STRINGIFY2(x)
+#define ARG_DESC_WIDEN2(x) L##x
+#define ARG_DESC_WIDEN(x) ARG_DESC_WIDEN2(x)
+#define LOCALIZED_NAME(name) ARG_DESC_WIDEN(ARG_DESC_STRINGIFY(name##Localized))
+
+// append this argument's description to the internal vector
+#define APPEND_ARG_DESCRIPTION(type, name, jsonKey, required, typeHint, ...) \
+    temp.push_back({ RS_(LOCALIZED_NAME(name)), L## #type, std::wstring_view(L## #required) != L"false", typeHint });

```

Comment:
(note that using `RS_` means it expands in the local language immediately on creation, rather than later during use)